### ISSUE TITLE
Default Keras callback for timing the training loop

### ISSUE THREAD

**Issue by Lokiiiiii** [keras-team/tf-keras#605]
If you open a GitHub issue, here is our policy:

It must be a bug, a feature request, or a significant problem with the documentation (for small docs fixes please send a PR instead).
The form below must be filled out.

**Here's why we have that policy:**.

Keras developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

**System information**.

TensorFlow version (you are using): nightly
Are you willing to contribute it (Yes/No) : Yes

**Describe the feature and the current behavior/state**.

The current Keras progress bar callback displays training latency in time/step. The current recommendation is to disable the progress bar in production workflows.

This poses 2 problems:

1. The metric time/step is not batch size agnostic. While tuning batch size to fit the H/W, metrics represented in terms of samples allow for better head to head comparison. Currently, an additional step of converting step to samples is required.

2. Production workflows that require continuous training will benefit greatly from being able to monitor training latency. A deterioration in training speed could lead to spike in training costs.

**Will this change the current api? How?**

No

**Who will benefit from this feature?**

Production workflows that require continuous training and need to monitor training speed.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes/no): Yes
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

1. Reporting an additional metric for training latency based on samples. Eg: samples/sec

2. A new verbosity setting that allows for reporting the training latency metrics without the progress bar. This could even be part of the existing verbose=0.
